nntrainer/models/model_loader.cpp

```diff
@@ -306,18 +306,20 @@ int ModelLoader::loadLayerConfigIniCommon(dictionary *ini,
   try {
     std::shared_ptr<ml::train::Layer> layer_ =
       app_context.createObject<ml::train::Layer>(layer_type, properties);
-    if (std::static_pointer_cast<Layer>(layer_)->getDistribute()) {
-      ml_logd("This %s layer is going to distributed",
-              layer_->getName().c_str());
-      std::shared_ptr<ml::train::Layer> dist_layer =
-        app_context.createObject<ml::train::Layer>(LayerType::LAYER_TIME_DIST);
-      layer = std::static_pointer_cast<Layer>(dist_layer);
-      std::shared_ptr<TimeDistLayer> drived =
-        std::dynamic_pointer_cast<TimeDistLayer>(layer);
-      drived->setDistLayer(std::static_pointer_cast<Layer>(layer_));
+    std::shared_ptr<Layer> l = std::static_pointer_cast<Layer>(layer_);
+
+    if (l->getDistribute()) {
+      ml_logd("This %s layer is going to distributed", layer_name.c_str());
+
+      std::shared_ptr<Layer> dist_layer =
+        nntrainer::createLayer(nntrainer::TimeDistLayer::type);
+
+      std::dynamic_pointer_cast<TimeDistLayer>(dist_layer)->setDistLayer(l);
```

Comment:
dynamic_cast breaks our api.

I think, later...., 1. we should put this into our api, 2. implement this with exisiting set of api, 3. introduce new object.

Reply:
> dynamic_cast breaks our api.

Could you give more specific?

Reply:
API does not have the method `setDistLayer` while we are using it as a part of layer.
We shouldn't be relying on the concrete type, but only the interfaces. If it's not possible, it suggests that there'd be some problem with our interface.

Reply:
Well we are not going to open this dist layer to the public. All the users have to do is add just dist_layer=true/false.

Reply:
If we are to define this is not part of an api, I think it's better to directly construct distlayer instead of creating `<Layer>` and dynamic_cast, also remove from the `appcontext`.